revup/config.py

```diff
@@ -68,34 +151,55 @@ def config_main(conf: Config, args: argparse.Namespace) -> int:
     else:
         raise RevupUsageException("Invalid flag argument (must be <key> or <command>.<key>)")
 
-    if not args.value:
-        value = getpass.getpass(f"Input value for {command}.{key}: ").strip()
-    else:
-        value = args.value
+    all_commands = {p.get_command(): p for p in all_parsers}
+    if command not in all_commands:
+        raise RevupUsageException(
+            f"Invalid command section {command}, choose from {list(all_commands.keys())}"
+        )
 
+    parser = all_commands[command]
+    actions = parser.get_actions()
+
+    if not args.delete:
+        if key not in actions:
+            raise RevupUsageException(
+                f"Invalid option key {key}, choose from {list(actions.keys())}"
+            )
+
+    if args.delete:
+        value = None
+        if args.value:
+            raise RevupUsageException("Can't provide a value when using --delete")
```

Comment:
This should probably mention which key+value has the issue, at least what the value is (e.g. if the problem is some extra arg that's unintentionally interpreted as a value or something)